### ISSUE TITLE
1.4 - XLM-RoBERTa fine-tuning

### ISSUE THREAD

**Issue by AriadnaHV** [AriadnaHV/Fighting-Hate-Speech#9]
- [x]  Configure XLM-RoBERTa for binary classification
- [x]  Define custom classification head with intermediate hidden layer (768 → 256 → 2)
- [x]  Hyperparameter tuning with Optuna (learning rate, batch size, number of epochs, warm-up steps)
- [x]  Fine-tune on training set
- [x]  Evaluate on validation and test sets
- [x]  Compare results with traditional ML baseline
- [x]  Save model to _sinodio-models_ bucket
